<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>PARANOiD Parameters &mdash; PARANOiD  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
        <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Included Analyses" href="ANALYSES.html" />
    <link rel="prev" title="Basic usage of PARANOiD" href="BASICUSAGE.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> PARANOiD
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="INPUT.html">Inputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="BASICUSAGE.html">Basic Usage</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Parameters</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#reads">--reads</a></li>
<li class="toctree-l2"><a class="reference internal" href="#barcodes">--barcodes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#reference">--reference</a></li>
<li class="toctree-l2"><a class="reference internal" href="#annotation">--annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#merge-replicates">--merge_replicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="#correlation-analysis">--correlation_analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#barcode-pattern">--barcode_pattern</a></li>
<li class="toctree-l2"><a class="reference internal" href="#domain">--domain</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">--output</a></li>
<li class="toctree-l2"><a class="reference internal" href="#min-length">--min_length</a></li>
<li class="toctree-l2"><a class="reference internal" href="#min-qual">--min_qual</a></li>
<li class="toctree-l2"><a class="reference internal" href="#min-percent-qual-filter">--min_percent_qual_filter</a></li>
<li class="toctree-l2"><a class="reference internal" href="#barcode-mismatches">--barcode_mismatches</a></li>
<li class="toctree-l2"><a class="reference internal" href="#mapq">--mapq</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#score-meanings-for-bowtie2-domain-pro">Score meanings for Bowtie2 (--domain pro)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#score-meanings-for-star-domain-eu">Score meanings for STAR (--domain eu)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#map-to-transcripts">--map_to_transcripts</a></li>
<li class="toctree-l2"><a class="reference internal" href="#number-top-transcripts">--number_top_transcripts</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-peak-calling">--omit_peak_calling</a></li>
<li class="toctree-l2"><a class="reference internal" href="#peak-calling-for-high-coverage">--peak_calling_for_high_coverage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#peak-calling-regions">--peak_calling_regions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#peak-calling-regions-width">--peak_calling_regions_width</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gene-id">--gene_id</a></li>
<li class="toctree-l2"><a class="reference internal" href="#color-barplot">--color_barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="#rna-subtypes">--rna_subtypes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-peak-distance">--omit_peak_distance</a></li>
<li class="toctree-l2"><a class="reference internal" href="#distance">--distance</a></li>
<li class="toctree-l2"><a class="reference internal" href="#percentile">--percentile</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-sequence-extraction">--omit_sequence_extraction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#seq-len">--seq_len</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-cl-nucleotide">--omit_cl_nucleotide</a></li>
<li class="toctree-l2"><a class="reference internal" href="#max-motif-num">--max_motif_num</a></li>
<li class="toctree-l2"><a class="reference internal" href="#min-motif-width">--min_motif_width</a></li>
<li class="toctree-l2"><a class="reference internal" href="#max-motif-width">--max_motif_width</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="ANALYSES.html">Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="OUTPUT.html">Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="ADDITIONAL_SCRIPTS.html">Additional scripts</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">FAQ</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PARANOiD</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a></li>
      <li class="breadcrumb-item active">PARANOiD Parameters</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/PARAMETERS.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="paranoid-parameters">
<h1>PARANOiD Parameters<a class="headerlink" href="#paranoid-parameters" title="Permalink to this heading"></a></h1>
<p>Explanation of all PARANOiD parameters</p>
<div class="section" id="reads">
<span id="parameters-read-file"></span><h2>--reads<a class="headerlink" href="#reads" title="Permalink to this heading"></a></h2>
<div class="line-block">
<div class="line">Essential parameter!</div>
<div class="line">States <a class="reference internal" href="INPUT.html#read-file"><span class="std std-ref">file</span></a> containing reads obtained by iCLIP experiments.</div>
<div class="line">Expects a <code class="docutils literal notranslate"><span class="pre">FASTQ</span></code> file.</div>
</div>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--reads /path/to/input-file.fastq
</pre></div>
</div>
</div>
<div class="section" id="barcodes">
<span id="parameters-barcodes"></span><h2>--barcodes<a class="headerlink" href="#barcodes" title="Permalink to this heading"></a></h2>
<div class="line-block">
<div class="line">Essential parameter!</div>
<div class="line">States <a class="reference internal" href="INPUT.html#barcodes"><span class="std std-ref">file</span></a> containing barcode sequences and experiment names. Necessary to split reads and allocate them to their experiment.</div>
<div class="line">Expects a <code class="docutils literal notranslate"><span class="pre">TSV</span></code> file.</div>
</div>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--barcodes /path/to/barcodes.tsv
</pre></div>
</div>
</div>
<div class="section" id="reference">
<span id="parameters-reference"></span><h2>--reference<a class="headerlink" href="#reference" title="Permalink to this heading"></a></h2>
<div class="line-block">
<div class="line">Essential parameter!</div>
<div class="line">States <a class="reference internal" href="INPUT.html#reference"><span class="std std-ref">reference genome</span></a> used to align reads to and thus to determine the location of cross-link sites.</div>
<div class="line">Expects a <code class="docutils literal notranslate"><span class="pre">FASTA</span></code> file.</div>
</div>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--reference /path/to/reference.fasta
</pre></div>
</div>
</div>
<div class="section" id="annotation">
<span id="parameters-annotation"></span><h2>--annotation<a class="headerlink" href="#annotation" title="Permalink to this heading"></a></h2>
<div class="line-block">
<div class="line">States <a class="reference internal" href="INPUT.html#annotation"><span class="std std-ref">annotation file</span></a> used for the <a class="reference internal" href="ANALYSES.html#rna-subtype-analysis"><span class="std std-ref">RNA subtype analysius</span></a>.</div>
<div class="line">Expects a <code class="docutils literal notranslate"><span class="pre">GFF</span></code> or <code class="docutils literal notranslate"><span class="pre">GTF</span></code> file.</div>
</div>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--annotation /path/to/annotation.gff
</pre></div>
</div>
</div>
<div class="section" id="merge-replicates">
<span id="id1"></span><h2>--merge_replicates<a class="headerlink" href="#merge-replicates" title="Permalink to this heading"></a></h2>
<p>Merges replicates into a single representatiove form. In order to do so experiment names need to named in a particular manner which is further explained in the  <a class="reference internal" href="INPUT.html#barcodes"><span class="std std-ref">barcodes section</span></a>.</p>
<p><code class="docutils literal notranslate"><span class="pre">Default:</span> <span class="pre">false</span></code></p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--merge_replicates
</pre></div>
</div>
</div>
<div class="section" id="correlation-analysis">
<span id="id2"></span><h2>--correlation_analysis<a class="headerlink" href="#correlation-analysis" title="Permalink to this heading"></a></h2>
<p>Only applies when <a class="reference internal" href="#merge-replicates"><span class="std std-ref">replicate merging</span></a> is chosen.
Does a correlation analysis of replicates to show their similarity (and thus if they should be merged at all).
Can cause problems with large reference genomes due to excessive RAM usage.</p>
<p><code class="docutils literal notranslate"><span class="pre">Default:</span> <span class="pre">false</span></code></p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--correlation_analysis
</pre></div>
</div>
</div>
<div class="section" id="barcode-pattern">
<span id="id3"></span><h2>--barcode_pattern<a class="headerlink" href="#barcode-pattern" title="Permalink to this heading"></a></h2>
<p>Adapt barcode patterns to different protocols. Default protocol is <a class="reference external" href="https://doi.org/10.1016/j.ymeth.2019.10.003">iCLIP2</a>.
<code class="docutils literal notranslate"><span class="pre">N</span></code> s represent the random barcode and <code class="docutils literal notranslate"><span class="pre">X</span></code> s the experimental barcode</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--barcode_pattern NNNNNXXXXXXNNNN
</pre></div>
</div>
<p>Example for iCLIP1</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--barcode_pattern NNNXXXXNN
</pre></div>
</div>
</div>
<div class="section" id="domain">
<span id="id4"></span><h2>--domain<a class="headerlink" href="#domain" title="Permalink to this heading"></a></h2>
<p>Choose between bowtie2 and STAR to be used to align reads to the reference sequence. Bowtie2 should be used for prokarytic organisms or transcript sequences while STAR should be used for eukaryotic organisms (or rather all splicing capable organisms) as STAR is splicing aware. If using STAR for splicing capable organisms it is highly recommended to provide an <a class="reference internal" href="INPUT.html#annotation"><span class="std std-ref">annotation file</span></a> file besides the reference.</p>
<p>Options:
| pro -&gt; Bowtie2 (default)
| eu  -&gt; STAR</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--domain pro
</pre></div>
</div>
</div>
<div class="section" id="output">
<span id="output-dir"></span><h2>--output<a class="headerlink" href="#output" title="Permalink to this heading"></a></h2>
<p>Specify directory to which output generated by PARANOiD will be written.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--output ./output
</pre></div>
</div>
</div>
<div class="section" id="min-length">
<span id="min-read-length"></span><h2>--min_length<a class="headerlink" href="#min-length" title="Permalink to this heading"></a></h2>
<p>Specify minimum length a read needs to have after adapter removal to persist. Reads that become shorter during adapter removal will be filtered out.</p>
<p>Usage (default)</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--min_length <span class="m">30</span>
</pre></div>
</div>
</div>
<div class="section" id="min-qual">
<span id="min-base-qual"></span><h2>--min_qual<a class="headerlink" href="#min-qual" title="Permalink to this heading"></a></h2>
<p>Minimum quality for bases. All bases below that quality are cut off.
The quality score (also known as Phred quality score) describes the certainty of correctness of the base and is typically calculated as follows with e being the error probability: <span class="math notranslate nohighlight">\(Q-Score = -10log_\text{10}(e)\)</span></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 37%" />
<col style="width: 33%" />
<col style="width: 30%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Phred Quality score</p></th>
<th class="head"><p>Error probability</p></th>
<th class="head"><p>Accuracy</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>10</p></td>
<td><p>10%</p></td>
<td><p>90%</p></td>
</tr>
<tr class="row-odd"><td><p>20</p></td>
<td><p>1%</p></td>
<td><p>99%</p></td>
</tr>
<tr class="row-even"><td><p>30</p></td>
<td><p>0.1%</p></td>
<td><p>99.9%</p></td>
</tr>
<tr class="row-odd"><td><p>40</p></td>
<td><p>0.01%</p></td>
<td><p>99.99%</p></td>
</tr>
</tbody>
</table>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--min_qual <span class="m">20</span>
</pre></div>
</div>
</div>
<div class="section" id="min-percent-qual-filter">
<span id="percent-qual-filter"></span><h2>--min_percent_qual_filter<a class="headerlink" href="#min-percent-qual-filter" title="Permalink to this heading"></a></h2>
<p>Percentage of nucleotides that need to have a quality score above the chosen <a class="reference internal" href="#min-base-qual"><span class="std std-ref">minimum base quality</span></a>.
Reads with less nucleotides above the desired quality will be removed.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--min_percent_qual_filter <span class="m">90</span>
</pre></div>
</div>
</div>
<div class="section" id="barcode-mismatches">
<span id="id5"></span><h2>--barcode_mismatches<a class="headerlink" href="#barcode-mismatches" title="Permalink to this heading"></a></h2>
<p>Number of mismatches allowed within the experimental barcode to still assign a read to an experiment.
Typically, experimental barcodes should be designed with a v of at least 3 to each other in order to allow one mismatch.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--barcode_mismatches <span class="m">1</span>
</pre></div>
</div>
</div>
<div class="section" id="mapq">
<span id="id6"></span><h2>--mapq<a class="headerlink" href="#mapq" title="Permalink to this heading"></a></h2>
<p>Minimum alignment quality (mapq score) an alignment needs to retain. The meaning of different scores is dependant on the aligner chosen via <a class="reference internal" href="#domain"><span class="std std-ref">--domain</span></a>.
All alignments with a mapq score below will be removed after the alignment step.
Please note that these are just a short overview of the meaning of MAPQ scores and that they can be more complex than shown here when going into details.
the MAPQ score can be found in alignment files (SAM/BAM/CRAM) in column 5.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--mapq <span class="m">2</span>
</pre></div>
</div>
<div class="section" id="score-meanings-for-bowtie2-domain-pro">
<h3>Score meanings for Bowtie2 (--domain pro)<a class="headerlink" href="#score-meanings-for-bowtie2-domain-pro" title="Permalink to this heading"></a></h3>
<p>Apart from the description in the table a higher MAPQ score means less allowed mismatches (with difference of the base quality a mismatched nucleotide has)</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 84%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>MAPQ score</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p>All mappable reads</p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p>Multimapped reads that have the same alignment quality at different positions</p></td>
</tr>
<tr class="row-even"><td><p>2-39</p></td>
<td><p>Mulitmapped reads that have one specific alignment with a better score than the other potential positions</p></td>
</tr>
<tr class="row-odd"><td><p>40</p></td>
<td><p>Reads mappable to only one position</p></td>
</tr>
<tr class="row-even"><td><p>42</p></td>
<td><p>Reads mappable to only one position with an almost perfect alignment. Best MAPQ score in Bowtie2 alignments</p></td>
</tr>
</tbody>
</table>
<p>More information can be found <a class="reference external" href="http://biofinysics.blogspot.com/2014/05/how-does-bowtie2-assign-mapq-scores.html">here</a></p>
</div>
<div class="section" id="score-meanings-for-star-domain-eu">
<h3>Score meanings for STAR (--domain eu)<a class="headerlink" href="#score-meanings-for-star-domain-eu" title="Permalink to this heading"></a></h3>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 84%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>MAPQ score</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p>Maps to 10 or more positions</p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p>Maps to 4-9 positions</p></td>
</tr>
<tr class="row-even"><td><p>2</p></td>
<td><p>Maps to 3 positions</p></td>
</tr>
<tr class="row-odd"><td><p>3</p></td>
<td><p>Maps to 2 positions</p></td>
</tr>
<tr class="row-even"><td><p>255</p></td>
<td><p>Reads mappable to only one position. Best MAPQ score in STAR alignments.</p></td>
</tr>
</tbody>
</table>
<p>The mapping quality MAPQ (column 5) is 255 for uniquely mapping reads, and  <span class="math notranslate nohighlight">\(MAPQ score = int(-10log_\text{10}(1-1/[\text{number of positions the read maps to}]))\)</span> for multi-mapping reads. This scheme is same as the one used by TopHat [...]
Source: <a class="reference external" href="https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf">Bowtie2 manual</a></p>
</div>
</div>
<div class="section" id="map-to-transcripts">
<span id="id7"></span><h2>--map_to_transcripts<a class="headerlink" href="#map-to-transcripts" title="Permalink to this heading"></a></h2>
<p>Should be used when transcripts are given as reference instead of a reference genome. Returns the transcripts with most hits from each sample.
More information can be found <a class="reference internal" href="ANALYSES.html#transcript-analysis"><span class="std std-ref">here</span></a></p>
<p><code class="docutils literal notranslate"><span class="pre">Default:</span> <span class="pre">false</span></code></p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--map_to_transcripts
</pre></div>
</div>
</div>
<div class="section" id="number-top-transcripts">
<span id="id8"></span><h2>--number_top_transcripts<a class="headerlink" href="#number-top-transcripts" title="Permalink to this heading"></a></h2>
<p>The number of transcripts with most hits that are selected from each sample if parameter <a class="reference internal" href="#map-to-transcripts"><span class="std std-ref">--map_to_transcripts</span></a> was used.
As the amount is chosen from each sample the total number of transcripts can excede this number.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--number_top_transcripts <span class="m">10</span>
</pre></div>
</div>
</div>
<div class="section" id="omit-peak-calling">
<span id="id9"></span><h2>--omit_peak_calling<a class="headerlink" href="#omit-peak-calling" title="Permalink to this heading"></a></h2>
<p>If specified <a class="reference internal" href="ANALYSES.html#peak-calling"><span class="std std-ref">peak calling</span></a> will not be performed.
Will be performed by default.</p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--omit_peak_calling
</pre></div>
</div>
</div>
<div class="section" id="peak-calling-for-high-coverage">
<span id="id10"></span><h2>--peak_calling_for_high_coverage<a class="headerlink" href="#peak-calling-for-high-coverage" title="Permalink to this heading"></a></h2>
<p>Only has an effect if <a class="reference internal" href="ANALYSES.html#peak-calling"><span class="std std-ref">peak calling</span></a> is performed.
Proteins covering the whole reference genome can cause problems for PureCLIP causing it to throw an error.
From our experience the parameters added by this argument can help PureCLIP with performing it's analysis.
Adds following arguments to the PureCLIP execution: <code class="docutils literal notranslate"><span class="pre">-mtc</span> <span class="pre">5000</span> <span class="pre">-mtc2</span> <span class="pre">5000</span> <span class="pre">-ld</span></code></p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--peak_calling_for_high_coverage
</pre></div>
</div>
</div>
<div class="section" id="peak-calling-regions">
<span id="id11"></span><h2>--peak_calling_regions<a class="headerlink" href="#peak-calling-regions" title="Permalink to this heading"></a></h2>
<p>Only has an effect if <a class="reference internal" href="ANALYSES.html#peak-calling"><span class="std std-ref">peak calling</span></a> is performed.
If specified peak regions instead of single peaks will be returned by PureCLIP.</p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--peak_calling_regions
</pre></div>
</div>
</div>
<div class="section" id="peak-calling-regions-width">
<span id="peak-calling-region-width"></span><h2>--peak_calling_regions_width<a class="headerlink" href="#peak-calling-regions-width" title="Permalink to this heading"></a></h2>
<p>Only has an effect if <a class="reference internal" href="#peak-calling-regions"><span class="std std-ref">peak calling regions</span></a> are stated.
Changes the width of peak calling regions returned by PureCLIP.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--peak_calling_regions_width <span class="m">8</span>
</pre></div>
</div>
</div>
<div class="section" id="gene-id">
<span id="id12"></span><h2>--gene_id<a class="headerlink" href="#gene-id" title="Permalink to this heading"></a></h2>
<p>Only has an effect if an <a class="reference internal" href="INPUT.html#annotation"><span class="std std-ref">annotation file</span></a> is provided and thus the <a class="reference internal" href="ANALYSES.html#rna-subtype-analysis"><span class="std std-ref">RNA subtype analysis</span></a> performed.
Wording of the tag that describes the gene ID. Is found in the last column of annotation files, typically as the first tag-value pair.
The column looks similar to this <code class="docutils literal notranslate"><span class="pre">ID=gene-LOC101842720;Dbxref=GeneID:101842720;Name=LOC101842720;gbkey=Gene;gene=LOC101842720;gene_biotype=pseudogene;pseudo=true</span></code>. In this case the tag necessary is <cite>ID</cite>.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--gene_id ID
</pre></div>
</div>
</div>
<div class="section" id="color-barplot">
<span id="id13"></span><h2>--color_barplot<a class="headerlink" href="#color-barplot" title="Permalink to this heading"></a></h2>
<p>Color of barplots returned by PARANOiD.
Affects graphs generated by <a class="reference internal" href="OUTPUT.html#output-peak-height-distribution"><span class="std std-ref">peak height distribution</span></a>, <a class="reference internal" href="ANALYSES.html#rna-subtype-analysis"><span class="std std-ref">RNA subtype analysis</span></a> and the <a class="reference internal" href="OUTPUT.html#output-experimental-barcode-distribution"><span class="std std-ref">experimental barcode distribution</span></a>.
Color is staded via a hexadecimal color code. If unsure which code translates to which color several websites can help to pick the correct one. <a class="reference external" href="https://www.color-hex.com/">Example</a></p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--color_barplot <span class="c1">#69b3a2</span>
</pre></div>
</div>
</div>
<div class="section" id="rna-subtypes">
<span id="id14"></span><h2>--rna_subtypes<a class="headerlink" href="#rna-subtypes" title="Permalink to this heading"></a></h2>
<p>Only has an effect if an <a class="reference internal" href="INPUT.html#annotation"><span class="std std-ref">annotation file</span></a> is provided and thus the <a class="reference internal" href="ANALYSES.html#rna-subtype-analysis"><span class="std std-ref">RNA subtype analysis</span></a> performed.
RNA subtypes/regions that shall be included in the <a class="reference internal" href="ANALYSES.html#rna-subtype-analysis"><span class="std std-ref">RNA subtype analysis</span></a>. RNA subtypes need to be separated by a <cite>,</cite> and should appear in the <a class="reference internal" href="INPUT.html#annotation"><span class="std std-ref">annotation file</span></a> within the <strong>feature type</strong> column (3rd column).
If both requirements are not met the analysis will either not be performed correctly or be aborted.
If not sure which RNA subtypes are included within your annotation file you can use the script <a class="reference internal" href="ADDITIONAL_SCRIPTS.html#determine-feature-types"><span class="std std-ref">featuretypes-from-gtfgff.awk</span></a>.
Additionally, users should beware not to choose subtypes/regions that are in a hierarchical relationship to each other as they can cover the same regions and thus make affected peaks appear as <strong>ambiguous</strong>.
Inormation about the hierarchical structure of RNA subtypes/regions can be obtained <a class="reference external" href="https://github.com/The-Sequence-Ontology/SO-Ontologies/blob/master/Ontology_Files/subsets/SOFA.obo/">here</a>.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--rna_subtypes 3_prime_UTR,transcript,5_prime_UTR
</pre></div>
</div>
</div>
<div class="section" id="omit-peak-distance">
<span id="id16"></span><h2>--omit_peak_distance<a class="headerlink" href="#omit-peak-distance" title="Permalink to this heading"></a></h2>
<p>Omits the <a class="reference internal" href="ANALYSES.html#peak-distance-analysis"><span class="std std-ref">peak distance analysis</span></a></p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--omit_peak_distance
</pre></div>
</div>
</div>
<div class="section" id="distance">
<span id="max-peak-distance"></span><h2>--distance<a class="headerlink" href="#distance" title="Permalink to this heading"></a></h2>
<p>Max distance used for the <a class="reference internal" href="ANALYSES.html#peak-distance-analysis"><span class="std std-ref">peak distance analysis</span></a>.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--distance <span class="m">30</span>
</pre></div>
</div>
</div>
<div class="section" id="percentile">
<span id="distance-percentile"></span><h2>--percentile<a class="headerlink" href="#percentile" title="Permalink to this heading"></a></h2>
<p>Peak percentiles for <a class="reference internal" href="ANALYSES.html#peak-distance-analysis"><span class="std std-ref">peak distance analysis</span></a> and <a class="reference internal" href="ANALYSES.html#motif-detection"><span class="std std-ref">sequence extraction/motif analysis</span></a>.
Only peaks with a value above this threshold are considered while all peaks below are omitted as background noise.
A percentile of 90 means that only top 10% of peaks are used.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--percentile <span class="m">90</span>
</pre></div>
</div>
</div>
<div class="section" id="omit-sequence-extraction">
<span id="id17"></span><h2>--omit_sequence_extraction<a class="headerlink" href="#omit-sequence-extraction" title="Permalink to this heading"></a></h2>
<p>Omits the <a class="reference internal" href="ANALYSES.html#motif-detection"><span class="std std-ref">motif detection</span></a></p>
<p>Usage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--omit_sequence_extraction
</pre></div>
</div>
</div>
<div class="section" id="seq-len">
<span id="sequence-extraction-length"></span><h2>--seq_len<a class="headerlink" href="#seq-len" title="Permalink to this heading"></a></h2>
<p>Only applies when <a class="reference internal" href="ANALYSES.html#motif-detection"><span class="std std-ref">motif detection</span></a> is performed.
Length in nucleotides to each side of a peak that is extracted from the <a class="reference internal" href="INPUT.html#reference"><span class="std std-ref">reference</span></a>.
A value of 20 will lead to sequences of 41 nucleotides being extracted. (20nt upstream;cross-link nt;20nt downstream)</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--seq_len <span class="m">20</span>
</pre></div>
</div>
</div>
<div class="section" id="omit-cl-nucleotide">
<span id="id18"></span><h2>--omit_cl_nucleotide<a class="headerlink" href="#omit-cl-nucleotide" title="Permalink to this heading"></a></h2>
<p>Only applies when <a class="reference internal" href="ANALYSES.html#motif-detection"><span class="std std-ref">motif detection</span></a> is performed.
The nucleotide directly at the cross-linking position will be substituted with an <strong>N</strong> when extracting sequences.
Can improve the motif detection since iCLIP tends to have a bias towards <strong>U</strong> when cross-linking which can influence the motif search.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--omit_cl_nucleotide
</pre></div>
</div>
</div>
<div class="section" id="max-motif-num">
<span id="max-number-of-motifs"></span><h2>--max_motif_num<a class="headerlink" href="#max-motif-num" title="Permalink to this heading"></a></h2>
<p>Only applies when <a class="reference internal" href="ANALYSES.html#motif-detection"><span class="std std-ref">motif detection</span></a> is performed.
Maximum number of motifs that is reported by streme.</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--max_motif_num <span class="m">50</span>
</pre></div>
</div>
</div>
<div class="section" id="min-motif-width">
<span id="id19"></span><h2>--min_motif_width<a class="headerlink" href="#min-motif-width" title="Permalink to this heading"></a></h2>
<p>Only applies when <a class="reference internal" href="ANALYSES.html#motif-detection"><span class="std std-ref">motif detection</span></a> is performed.
Minimum length of motifs reported by streme.
Cannot be lower than 3</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--min_motif_width <span class="m">8</span>
</pre></div>
</div>
</div>
<div class="section" id="max-motif-width">
<span id="id20"></span><h2>--max_motif_width<a class="headerlink" href="#max-motif-width" title="Permalink to this heading"></a></h2>
<p>Only applies when <a class="reference internal" href="ANALYSES.html#motif-detection"><span class="std std-ref">motif detection</span></a> is performed.
Maximum length of motifs reported by streme.
Cannot be higher than 30</p>
<p>Usage (default):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--max_motif_width <span class="m">15</span>
</pre></div>
</div>
</div>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="BASICUSAGE.html" class="btn btn-neutral float-left" title="Basic usage of PARANOiD" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="ANALYSES.html" class="btn btn-neutral float-right" title="Included Analyses" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Patrick Barth.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>