<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>PARANOiD Parameters &mdash; PARANOiD  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
        <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Included Analyses" href="ANALYSES.html" />
    <link rel="prev" title="PARANOiD Inputs" href="INPUT.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> PARANOiD
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="INPUT.html">Inputs</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Parameters</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#merge-replicates">Merge replicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="#correlation-analysis">Correlation analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#change-barcode-pattern">Change barcode pattern</a></li>
<li class="toctree-l2"><a class="reference internal" href="#choose-aligner">Choose aligner</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output-directory">Output directory</a></li>
<li class="toctree-l2"><a class="reference internal" href="#minimum-read-length-after-adapter-removal">Minimum read length after adapter removal</a></li>
<li class="toctree-l2"><a class="reference internal" href="#minimum-base-quality">Minimum base quality</a></li>
<li class="toctree-l2"><a class="reference internal" href="#minimum-percent-of-nucleotides-above-quality-threshold">Minimum percent of nucleotides above quality threshold</a></li>
<li class="toctree-l2"><a class="reference internal" href="#mismatches-allowed-within-barcodes">Mismatches allowed within barcodes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#alignment-quality">Alignment quality</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#score-meanings-for-bowtie2-domain-pro">Score meanings for Bowtie2 (–domain pro)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#score-meanings-for-star-domain-eu">Score meanings for STAR (–domain eu)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#map-to-transcripts">Map to transcripts</a></li>
<li class="toctree-l2"><a class="reference internal" href="#number-of-top-transcripts">Number of top transcripts</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-peak-calling">Omit peak calling</a></li>
<li class="toctree-l2"><a class="reference internal" href="#peak-calling-for-references-with-a-generally-high-coverage">Peak calling for references with a generally high coverage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#return-peak-regions-instead-of-single-peaks">Return peak regions instead of single peaks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#adapt-width-of-peak-regions">Adapt width of peak regions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#name-of-gene-id-in-annotation-file">Name of gene ID in annotation file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#color-of-barplot">Color of barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="#rna-subtypes">RNA subtypes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-peak-distance-analysis">Omit peak distance analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#distance-for-peak-distance-analysis">Distance for peak distance analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#peak-percentiles-for-peak-distance-analysis-and-sequence-extraction">Peak percentiles for peak distance analysis and sequence extraction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-sequence-extraction-and-motif-analysis">Omit sequence extraction and motif analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#length-for-sequence-extraction">Length for sequence extraction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#omit-nucleotide-at-cross-linking-site">Omit nucleotide at cross linking site</a></li>
<li class="toctree-l2"><a class="reference internal" href="#maximum-number-of-motifs">Maximum number of motifs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#minumum-width-of-motifs">Minumum width of motifs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#maximum-width-of-motifs">Maximum width of motifs</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="ANALYSES.html">Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="OUTPUT.html">Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="ADDITIONAL_SCRIPTS.html">Additional scripts</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">FAQ</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PARANOiD</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a></li>
      <li class="breadcrumb-item active">PARANOiD Parameters</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/PARAMETERS.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="paranoid-parameters">
<h1>PARANOiD Parameters<a class="headerlink" href="#paranoid-parameters" title="Permalink to this heading"></a></h1>
<p>Explanation of all PARANOiD parameters</p>
<div class="section" id="merge-replicates">
<span id="id1"></span><h2>Merge replicates<a class="headerlink" href="#merge-replicates" title="Permalink to this heading"></a></h2>
<p>Merges replicates into a single representatiove form. In order to do so experiment names need to named in a particular manner which is further explained in the  <a class="reference internal" href="INPUT.html#barcodes"><span class="std std-ref">[barcodes section]</span></a>.</p>
<p><code class="docutils literal notranslate"><span class="pre">Default:</span> <span class="pre">false</span></code></p>
<p>Usage:
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--merge_replicates</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="correlation-analysis">
<span id="id2"></span><h2>Correlation analysis<a class="headerlink" href="#correlation-analysis" title="Permalink to this heading"></a></h2>
<p>Only applies when <a class="reference internal" href="#merge-replicates"><span class="std std-ref">[replicate merging]</span></a> is chosen.
Does a correlation analysis of replicates to show their similarity (and thus if they should be merged at all).
Can cause problems with large reference genomes due to excessive RAM usage.</p>
<p><code class="docutils literal notranslate"><span class="pre">Default:</span> <span class="pre">false</span></code></p>
<p>Usage:
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--correlation_analysis</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="change-barcode-pattern">
<span id="barcode-pattern"></span><h2>Change barcode pattern<a class="headerlink" href="#change-barcode-pattern" title="Permalink to this heading"></a></h2>
<p>Adapt barcode patterns to different protocols. Default protocol is <a class="reference external" href="https://doi.org/10.1016/j.ymeth.2019.10.003">iCLIP2</a>.
<code class="docutils literal notranslate"><span class="pre">N</span></code> s represent the random barcode and <code class="docutils literal notranslate"><span class="pre">X</span></code> s the experimental barcode</p>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--barcode_pattern</span> <span class="pre">NNNNNXXXXXXNNNN</span>
<span class="pre">`</span></code>
Example for iCLIP1
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--barcode_pattern</span> <span class="pre">NNNXXXXNN</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="choose-aligner">
<span id="domain"></span><h2>Choose aligner<a class="headerlink" href="#choose-aligner" title="Permalink to this heading"></a></h2>
<p>Choose between bowtie2 and STAR to be used to align reads to the reference sequence. Bowtie2 should be used for prokarytic organisms or transcript sequences while STAR should be used for eukaryotic organisms (or rather all splicing capable organisms) as STAR is splicing aware. If using STAR for splicing capable organisms it is highly recommended to provide an <a class="reference internal" href="INPUT.html#annotation"><span class="std std-ref">[annotation file]</span></a> file besides the reference.</p>
<p>Options:
| pro -&gt; Bowtie2 (default)
| eu  -&gt; STAR</p>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--domain</span> <span class="pre">pro</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="output-directory">
<span id="output-dir"></span><h2>Output directory<a class="headerlink" href="#output-directory" title="Permalink to this heading"></a></h2>
<p>Specify directory to which output generated by PARANOiD will be written.</p>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--output</span> <span class="pre">./output</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="minimum-read-length-after-adapter-removal">
<span id="min-read-length"></span><h2>Minimum read length after adapter removal<a class="headerlink" href="#minimum-read-length-after-adapter-removal" title="Permalink to this heading"></a></h2>
<p>Specify minimum length a read needs to have after adapter removal to persist. Reads that become shorter during adapter removal will be filtered out.</p>
<p>Usage (default)
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_length</span> <span class="pre">30</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="minimum-base-quality">
<span id="min-base-qual"></span><h2>Minimum base quality<a class="headerlink" href="#minimum-base-quality" title="Permalink to this heading"></a></h2>
<p>Minimum quality for bases. All bases below that quality are cut off.
The quality score (also known as Phred quality score) describes the certainty of correctness of the base and is typically calculated as follows with e being the error probability: <span class="math notranslate nohighlight">\(Q-Score = -10log_\text{10}(e)\)</span></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 37%" />
<col style="width: 33%" />
<col style="width: 30%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Phred Quality score</p></th>
<th class="head"><p>Error probability</p></th>
<th class="head"><p>Accuracy</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>10</p></td>
<td><p>10%</p></td>
<td><p>90%</p></td>
</tr>
<tr class="row-odd"><td><p>20</p></td>
<td><p>1%</p></td>
<td><p>99%</p></td>
</tr>
<tr class="row-even"><td><p>30</p></td>
<td><p>0.1%</p></td>
<td><p>99.9%</p></td>
</tr>
<tr class="row-odd"><td><p>40</p></td>
<td><p>0.01%</p></td>
<td><p>99.99%</p></td>
</tr>
</tbody>
</table>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="minimum-percent-of-nucleotides-above-quality-threshold">
<span id="percent-qual-filter"></span><h2>Minimum percent of nucleotides above quality threshold<a class="headerlink" href="#minimum-percent-of-nucleotides-above-quality-threshold" title="Permalink to this heading"></a></h2>
<p>Percentage of nucleotides that need to have a quality score above the chosen <a class="reference internal" href="#min-base-qual"><span class="std std-ref">[minimum base quality]</span></a>.
Reads with less nucleotides above the desired quality will be removed.</p>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_percent_qual_filter</span> <span class="pre">90</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="mismatches-allowed-within-barcodes">
<span id="barcode-mismatches"></span><h2>Mismatches allowed within barcodes<a class="headerlink" href="#mismatches-allowed-within-barcodes" title="Permalink to this heading"></a></h2>
<p>Number of mismatches allowed within the experimental barcode to still assign a read to an experiment.
Typically, experimental barcodes should be designed with a v of at least 3 to each other in order to allow one mismatch.</p>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--barcode_mismatches</span> <span class="pre">1</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="alignment-quality">
<span id="mapq"></span><h2>Alignment quality<a class="headerlink" href="#alignment-quality" title="Permalink to this heading"></a></h2>
<p>Minimum alignment quality (mapq score) an alignment needs to retain. The meaning of different scores is dependant on the aligner chosen via <a class="reference internal" href="#domain"><span class="std std-ref">[–domain]</span></a>.
All alignments with a mapq score below will be removed after the alignment step.
Please note that these are just a short overview of the meaning of MAPQ scores and that they can be more complex than shown here when going into details.
the MAPQ score can be found in alignment files (SAM/BAM/CRAM) in column 5.</p>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--mapq</span> <span class="pre">2</span>
<span class="pre">`</span></code></p>
<div class="section" id="score-meanings-for-bowtie2-domain-pro">
<h3>Score meanings for Bowtie2 (–domain pro)<a class="headerlink" href="#score-meanings-for-bowtie2-domain-pro" title="Permalink to this heading"></a></h3>
<p>Apart from the description in the table a higher MAPQ score means less allowed mismatches (with difference of the base quality a mismatched nucleotide has)</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 84%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>MAPQ score</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p>All mappable reads</p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p>Multimapped reads that have the same alignment quality at different positions</p></td>
</tr>
<tr class="row-even"><td><p>2-39</p></td>
<td><p>Mulitmapped reads that have one specific alignment with a better score than the other potential positions</p></td>
</tr>
<tr class="row-odd"><td><p>40</p></td>
<td><p>Reads mappable to only one position</p></td>
</tr>
<tr class="row-even"><td><p>42</p></td>
<td><p>Reads mappable to only one position with an almost perfect alignment. Best MAPQ score in Bowtie2 alignments</p></td>
</tr>
</tbody>
</table>
<p>More information can be found <a class="reference external" href="http://biofinysics.blogspot.com/2014/05/how-does-bowtie2-assign-mapq-scores.html">here</a></p>
</div>
<div class="section" id="score-meanings-for-star-domain-eu">
<h3>Score meanings for STAR (–domain eu)<a class="headerlink" href="#score-meanings-for-star-domain-eu" title="Permalink to this heading"></a></h3>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 84%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>MAPQ score</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p>Maps to 10 or more positions</p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p>Maps to 4-9 positions</p></td>
</tr>
<tr class="row-even"><td><p>2</p></td>
<td><p>Maps to 3 positions</p></td>
</tr>
<tr class="row-odd"><td><p>3</p></td>
<td><p>Maps to 2 positions</p></td>
</tr>
<tr class="row-even"><td><p>255</p></td>
<td><p>Reads mappable to only one position. Best MAPQ score in STAR alignments.</p></td>
</tr>
</tbody>
</table>
<p>The mapping quality MAPQ (column 5) is 255 for uniquely mapping reads, and  <span class="math notranslate nohighlight">\(MAPQ score = int(-10log_\text{10}(1-1/[\text{number of positions the read maps to}]))\)</span> for multi-mapping reads. This scheme is same as the one used by TopHat […]
Source: <a class="reference external" href="https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf">https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf</a></p>
</div>
</div>
<div class="section" id="map-to-transcripts">
<span id="id3"></span><h2>Map to transcripts<a class="headerlink" href="#map-to-transcripts" title="Permalink to this heading"></a></h2>
<p>Should be used when transcripts are given as reference instead of a reference genome. Returns the transcripts with most hits from each sample.
More information can be found <span class="xref std std-ref">[here]</span></p>
<p><code class="docutils literal notranslate"><span class="pre">Default:</span> <span class="pre">false</span></code></p>
<p>Usage:
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--map_to_transcripts</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="number-of-top-transcripts">
<span id="number-top-transcripts"></span><h2>Number of top transcripts<a class="headerlink" href="#number-of-top-transcripts" title="Permalink to this heading"></a></h2>
<p>The number of transcripts with most hits that are selected from each sample if parameter <a class="reference internal" href="#map-to-transcripts"><span class="std std-ref">[–map_to_transcripts]</span></a> was used.
As the amount is chosen from each sample the total number of transcripts can excede this number.</p>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--number_top_transcripts</span> <span class="pre">10</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="omit-peak-calling">
<span id="id4"></span><h2>Omit peak calling<a class="headerlink" href="#omit-peak-calling" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="peak-calling-for-references-with-a-generally-high-coverage">
<span id="peak-calling-for-high-coverage"></span><h2>Peak calling for references with a generally high coverage<a class="headerlink" href="#peak-calling-for-references-with-a-generally-high-coverage" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="return-peak-regions-instead-of-single-peaks">
<span id="peak-calling-regions"></span><h2>Return peak regions instead of single peaks<a class="headerlink" href="#return-peak-regions-instead-of-single-peaks" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="adapt-width-of-peak-regions">
<span id="peak-calling-region-width"></span><h2>Adapt width of peak regions<a class="headerlink" href="#adapt-width-of-peak-regions" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="name-of-gene-id-in-annotation-file">
<span id="gene-id"></span><h2>Name of gene ID in annotation file<a class="headerlink" href="#name-of-gene-id-in-annotation-file" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="color-of-barplot">
<span id="color-barplot"></span><h2>Color of barplot<a class="headerlink" href="#color-of-barplot" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="rna-subtypes">
<span id="id5"></span><h2>RNA subtypes<a class="headerlink" href="#rna-subtypes" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="omit-peak-distance-analysis">
<span id="omit-peak-distance"></span><h2>Omit peak distance analysis<a class="headerlink" href="#omit-peak-distance-analysis" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="distance-for-peak-distance-analysis">
<span id="max-peak-distance"></span><h2>Distance for peak distance analysis<a class="headerlink" href="#distance-for-peak-distance-analysis" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="peak-percentiles-for-peak-distance-analysis-and-sequence-extraction">
<span id="distance-percentile"></span><h2>Peak percentiles for peak distance analysis and sequence extraction<a class="headerlink" href="#peak-percentiles-for-peak-distance-analysis-and-sequence-extraction" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="omit-sequence-extraction-and-motif-analysis">
<span id="omit-sequence-extraction"></span><h2>Omit sequence extraction and motif analysis<a class="headerlink" href="#omit-sequence-extraction-and-motif-analysis" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="length-for-sequence-extraction">
<span id="sequence-extarction-length"></span><h2>Length for sequence extraction<a class="headerlink" href="#length-for-sequence-extraction" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="omit-nucleotide-at-cross-linking-site">
<span id="omit-cl-nuckeotide"></span><h2>Omit nucleotide at cross linking site<a class="headerlink" href="#omit-nucleotide-at-cross-linking-site" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="maximum-number-of-motifs">
<span id="max-number-of-motifs"></span><h2>Maximum number of motifs<a class="headerlink" href="#maximum-number-of-motifs" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="minumum-width-of-motifs">
<span id="min-motif-width"></span><h2>Minumum width of motifs<a class="headerlink" href="#minumum-width-of-motifs" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
<div class="section" id="maximum-width-of-motifs">
<span id="max-motif-width"></span><h2>Maximum width of motifs<a class="headerlink" href="#maximum-width-of-motifs" title="Permalink to this heading"></a></h2>
<p>Usage (default):
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">--min_qual</span> <span class="pre">20</span>
<span class="pre">`</span></code></p>
</div>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="INPUT.html" class="btn btn-neutral float-left" title="PARANOiD Inputs" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="ANALYSES.html" class="btn btn-neutral float-right" title="Included Analyses" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Patrick Barth.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>