Search.setIndex({"docnames": ["ADDITIONAL_SCRIPTS", "ANALYSES", "FAQ", "INPUT", "OUTPUT", "PARAMETERS", "index"], "filenames": ["ADDITIONAL_SCRIPTS.rst", "ANALYSES.rst", "FAQ.rst", "INPUT.rst", "OUTPUT.rst", "PARAMETERS.rst", "index.rst"], "titles": ["Supplementary scripts for PARANOiD", "Included Analyses", "Frequently Asked Questions", "PARANOiD Inputs", "PARANOiD Inputs", "PARANOiD Parameters", "Welcome to PARANOiD\u2019s documentation!"], "terms": {"ad": [0, 1, 3], "analysi": [0, 3, 6], "name": [0, 3, 6], "featuretyp": 0, "from": [0, 3], "gtfgff": 0, "awk": 0, "usag": [0, 5], "path": 0, "anntotation_fil": 0, "gff": [0, 3], "short": 1, "overview": 1, "all": [0, 1, 3, 4, 5], "implement": 1, "paranoid": 1, "determin": [1, 6], "protein": 1, "interest": [1, 3], "i": [1, 3, 5], "prone": 1, "bind": 1, "specif": 1, "region": [1, 3, 6], "As": 1, "thi": [1, 3], "via": [0, 1, 3], "annot": [1, 6], "file": [1, 3, 6], "onli": [1, 3, 5], "can": [1, 3, 5], "shown": 1, "column": [1, 3], "3": [1, 3], "To": 1, "see": 1, "which": [1, 3, 5], "ar": [0, 1, 3, 5], "script": [1, 6], "wa": 1, "activ": 1, "when": [1, 3, 5], "an": [1, 3, 5], "provid": [1, 3, 5], "detail": 3, "descript": 3, "fastq": 3, "contain": [0, 1, 3], "each": [1, 3], "repres": [1, 3, 5], "4": 3, "line": 3, "sequenc": [1, 3, 6], "identifi": 3, "option": [3, 5], "start": [0, 3], "actual": 3, "nucleotid": [3, 6], "delimit": 3, "qualiti": [3, 6], "valu": 3, "line2": 3, "must": 3, "same": [1, 3], "number": [3, 6], "symbol": 3, "2": 3, "head": [], "exampl": [3, 5], "3744fb5d4a29f1e45b707323c09701aee51d2d4f": [], "seq_id": 3, "gatttggggttcaaagcagtatcgatcaaatagtaaatccatttgttcaactcacagttt": 3, "1": [3, 5], "55ccf": 3, "ccccccc65": 3, "tsv": 3, "experi": [3, 5], "correspond": 3, "split": 3, "accord": 3, "detect": [3, 6], "assign": 3, "appropri": 3, "result": 3, "one": [1, 3], "per": 3, "choos": [1, 3, 6], "merg": [3, 6], "replic": [3, 6], "merge_repl": [1, 3, 5], "have": [3, 5], "chosen": [3, 5], "indic": 3, "belong": 3, "togeth": 3, "In": [3, 5], "order": [3, 5], "do": [3, 5], "appendix": 3, "_rep_": 3, "ha": [1, 3], "exchang": 3, "consist": 3, "separ": 3, "singl": [1, 3, 6], "tab": 3, "present": [1, 3], "should": [0, 1, 3, 5], "letter": 3, "za": 3, "z": 3, "9": [3, 5], "underscor": 3, "_": 3, "ani": 3, "whitespac": 3, "e": [3, 5], "g": 3, "space": 3, "error": [3, 5], "thu": [3, 5], "termin": 3, "pipelin": 3, "execut": 3, "The": [1, 3, 5], "length": [3, 6], "depend": 3, "protocol": [3, 5], "us": [0, 1, 3, 5], "adapt": [3, 6], "barcode_pattern": [3, 5], "knockdown_n_rep_1": 3, "tgatag": 3, "knockdown_n_rep_2": 3, "agtgga": 3, "knockdown_n_rep_3": 3, "gctcga": 3, "mock_n_rep_1": 3, "taagta": 3, "mock_n_rep_2": 3, "gcagtc": 3, "mock_n_rep_3": 3, "cctagg": 3, "fasta": [1, 3], "data": 3, "align": [3, 6], "find": 3, "locat": 3, "cross": [3, 6], "link": [3, 6], "site": [3, 6], "genom": [1, 3, 5], "transcriptom": [1, 3], "organ": [1, 3, 5], "complet": 3, "artifici": [1, 3], "everi": [1, 3], "least": [1, 3], "header": 3, "n": [3, 5], "follow": [1, 3, 5], "acgtn": 3, "span": 3, "arbitrari": 3, "amount": 3, "nw_024429180": 3, "mesocricetu": 3, "auratu": 3, "isol": 3, "sy011": 3, "unplac": 3, "scaffold": 3, "aactctgttgtaaaaaggctttcccacattcattccattcataaggtttctgtacattatggattctttcatgcctttta": 3, "agatgattatgatatacatagactttaacacctcaagaatattcaggtttctctccagtatgacaatttggtctaattat": 3, "aaagaagaatcagatattaaggttttatcactgtttacactcatgctgttccccttcattaaggttggtttggatctttg": 3, "aatatacctgggttcctatagtctccaccatcacatctttatggagattcttctgggagggatccagcaaatcccactct": 3, "gtf": 3, "inform": 3, "describ": [3, 5], "featur": 3, "posit": [1, 3], "doe": [3, 5], "reli": 3, "": [3, 5], "howev": 3, "highli": [3, 5], "recommend": [3, 5], "work": 3, "splice": [3, 5], "capabl": [3, 5], "domain": [3, 5], "eu": [3, 5], "typic": [3, 5], "about": 3, "intron": 3, "exon": 3, "structur": [1, 3], "improv": 3, "map": 3, "furthermor": 3, "enabl": 3, "rna": [3, 6], "subtyp": [3, 6], "sever": [1, 3], "gener": [1, 3, 4, 6], "seqnam": 3, "chromosom": 3, "sourc": 3, "program": 3, "type": 3, "current": 3, "cd": 3, "mrna": 3, "3_prime_utr": 3, "transcript": [3, 6], "5_prime_utr": 3, "base": [3, 6], "5": 3, "end": 3, "6": 3, "score": [3, 5], "float": 3, "point": 3, "also": [3, 5], "simpli": 3, "7": 3, "strand": 3, "stradn": 3, "forward": 3, "revers": 3, "8": 3, "frame": 3, "first": 3, "codon": 3, "0": [3, 5], "abs": 3, "second": 3, "slo": 3, "attribut": 3, "semicolon": 3, "list": 3, "tag": 3, "pair": 3, "addit": [3, 6], "version": [1, 3], "spec": 3, "21": 3, "processor": 3, "ncbi": 3, "annotwrit": 3, "build": [0, 3], "bcm_maur_2": 3, "access": [1, 3], "ncbi_assembli": 3, "gcf_017639785": 3, "releas": 3, "103": 3, "52462669": 3, "speci": 3, "http": [1, 3, 5], "www": 3, "nlm": 3, "nih": 3, "gov": 3, "taxonomi": 3, "browser": 3, "wwwtax": 3, "cgi": 3, "id": [3, 6], "10036": 3, "refseq": 3, "dbxref": 3, "taxon": 3, "unknown": 3, "dev": 3, "stage": 3, "adult": 3, "gbkei": 3, "src": 3, "mol_typ": 3, "dna": 3, "sex": 3, "femal": 3, "tissu": 3, "liver": 3, "gnomon": 3, "pseudogen": 3, "37366": 3, "38359": 3, "gene": [3, 6], "loc101842720": 3, "geneid": 3, "101842720": 3, "gene_biotyp": 3, "pseudo": 3, "true": 3, "explan": [4, 5], "output": [4, 6], "representatiov": 5, "form": 5, "so": [1, 5], "need": [1, 5], "particular": 5, "manner": 5, "further": 5, "explain": 5, "section": 5, "default": [1, 5], "fals": 5, "differ": 5, "iclip2": 5, "doi": 5, "org": 5, "10": 5, "1016": 5, "j": 5, "ymeth": 5, "2019": 5, "003": 5, "random": 5, "x": 5, "experiment": 5, "nnnnnxxxxxxnnnn": 5, "iclip1": 5, "nnnxxxxnn": 5, "between": 5, "bowtie2": 5, "star": 5, "refer": [1, 6], "prokaryt": 5, "while": 5, "eukaryot": 5, "rather": 5, "awar": 5, "If": [1, 5], "besid": 5, "pro": 5, "specifi": 5, "written": 5, "persist": 5, "becom": 5, "shorter": 5, "dure": 5, "filter": 5, "out": 5, "min_length": 5, "30": 5, "input": [1, 6], "read": 6, "barcod": 6, "paramet": 6, "chang": 6, "pattern": 6, "directori": 6, "minimum": 6, "after": 6, "remov": 6, "analys": 6, "valid": 6, "faq": 6, "singular": 0, "process": 0, "befor": 0, "pull_imag": 0, "sh": 0, "dockerfil": 0, "image_directori": 0, "public": 1, "poster": 1, "show": [1, 5], "mean": 1, "hit": 1, "count": 1, "addition": 1, "correl": [1, 6], "perform": 1, "give": 1, "user": 1, "evalu": 1, "sampl": 1, "similar": [1, 5], "therefor": 1, "rational": 1, "deactiv": 1, "care": 1, "hierarch": 1, "higher": 1, "lower": 1, "other": 1, "partial": 1, "cover": 1, "regoin": 1, "make": 1, "ambigu": 1, "ontologi": 1, "github": 1, "com": 1, "blob": 1, "master": 1, "ontology_fil": 1, "subset": 1, "sofa": 1, "obo": 1, "get": 1, "offici": 1, "more": 1, "interact": 1, "instead": [1, 6], "here": 1, "combin": 1, "shall": 1, "examin": 1, "thei": [1, 5], "often": 1, "next": 1, "appli": 5, "caus": 5, "problem": 5, "larg": 5, "due": 5, "excess": 5, "ram": 5, "correlation_analysi": 5, "below": 5, "cut": 5, "off": 5, "known": 5, "phred": 5, "certainti": 5, "correct": 5, "calcul": 5, "being": 5, "probabl": 5, "q": 5, "10log_": 5, "text": 5, "accuraci": 5, "90": 5, "20": 5, "99": 5, "40": 5, "01": 5, "min_qual": 5, "percent": 6, "abov": 6, "threshold": 6, "mismatch": 6, "allow": 6, "within": 6, "top": 6, "omit": 6, "peak": 6, "call": 6, "high": 6, "coverag": 6, "return": 6, "width": 6, "color": 6, "barplot": 6, "distanc": 6, "percentil": 6, "extract": 6, "motif": 6, "maximum": 6, "minumum": 6, "pull": 6, "imag": 6, "map_to_transcript": 1}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"supplementari": 0, "script": 0, "paranoid": [0, 3, 4, 5, 6], "determin": 0, "valid": 0, "rna": [0, 1, 5], "subtyp": [0, 1, 5], "includ": 1, "analys": 1, "frequent": 2, "ask": 2, "question": 2, "input": [3, 4], "read": [3, 5], "barcod": [3, 5], "refer": [3, 5], "annot": [3, 5], "paramet": 5, "merg": [1, 5], "replic": [1, 5], "chang": 5, "pattern": 5, "choos": 5, "align": 5, "output": 5, "directori": 5, "minimum": 5, "length": 5, "after": 5, "adapt": 5, "remov": 5, "base": 5, "qualiti": 5, "welcom": 6, "": 6, "document": 6, "content": 6, "pull": 0, "imag": 0, "transcript": [1, 5], "analysi": [1, 5], "peak": [1, 5], "call": [1, 5], "distanc": [1, 5], "motif": [1, 5], "detect": 1, "correl": 5, "percent": 5, "nucleotid": 5, "abov": 5, "threshold": 5, "mismatch": 5, "allow": 5, "within": 5, "number": 5, "top": 5, "omit": 5, "gener": 5, "high": 5, "coverag": 5, "return": 5, "region": 5, "instead": 5, "singl": 5, "width": 5, "name": 5, "gene": 5, "id": 5, "file": 5, "color": 5, "barplot": 5, "percentil": 5, "sequenc": 5, "extract": 5, "cross": 5, "link": 5, "site": 5, "maximum": 5, "minumum": 5}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"Frequently Asked Questions": [[2, "frequently-asked-questions"]], "PARANOiD Inputs": [[4, "paranoid-inputs"], [3, "paranoid-inputs"]], "Supplementary scripts for PARANOiD": [[0, "supplementary-scripts-for-paranoid"]], "Determine valid RNA subtypes": [[0, "determine-valid-rna-subtypes"]], "Pull images": [[0, "pull-images"]], "Reads": [[3, "reads"]], "Barcodes": [[3, "barcodes"]], "Reference": [[3, "reference"]], "Annotation": [[3, "annotation"]], "Welcome to PARANOiD\u2019s documentation!": [[6, "welcome-to-paranoid-s-documentation"]], "Contents:": [[6, null]], "Merge replicates": [[1, "merge-replicates"], [5, "merge-replicates"]], "RNA subtypes": [[1, "rna-subtypes"], [5, "id4"]], "Included Analyses": [[1, "included-analyses"]], "Transcript analysis": [[1, "transcript-analysis"]], "Peak calling": [[1, "peak-calling"]], "Peak distance analysis": [[1, "peak-distance-analysis"]], "Motif detection": [[1, "motif-detection"]], "PARANOiD Parameters": [[5, "paranoid-parameters"]], "Correlation analysis": [[5, "correlation-analysis"]], "Change barcode pattern": [[5, "change-barcode-pattern"]], "Choose aligner": [[5, "choose-aligner"]], "Output directory": [[5, "output-directory"]], "Minimum read length after adapter removal": [[5, "minimum-read-length-after-adapter-removal"]], "Minimum base quality": [[5, "minimum-base-quality"]], "Minimum percent of nucleotides above quality threshold": [[5, "minimum-percent-of-nucleotides-above-quality-threshold"]], "Mismatches allowed within barcodes": [[5, "mismatches-allowed-within-barcodes"]], "Alignment quality": [[5, "alignment-quality"]], "Align to transcripts": [[5, "align-to-transcripts"]], "Number of top transcripts": [[5, "number-of-top-transcripts"]], "Omit peak calling": [[5, "id3"]], "Peak calling for references with a generally high coverage": [[5, "peak-calling-for-references-with-a-generally-high-coverage"]], "Return peak regions instead of single peaks": [[5, "return-peak-regions-instead-of-single-peaks"]], "Adapt width of peak regions": [[5, "adapt-width-of-peak-regions"]], "Name of gene ID in annotation file": [[5, "name-of-gene-id-in-annotation-file"]], "Color of barplot": [[5, "color-of-barplot"]], "Omit peak distance analysis": [[5, "omit-peak-distance-analysis"]], "Distance for peak distance analysis": [[5, "distance-for-peak-distance-analysis"]], "Peak percentiles for peak distance analysis and sequence extraction": [[5, "peak-percentiles-for-peak-distance-analysis-and-sequence-extraction"]], "Omit sequence extraction and motif analysis": [[5, "omit-sequence-extraction-and-motif-analysis"]], "Length for sequence extraction": [[5, "length-for-sequence-extraction"]], "Omit nucleotide at cross linking site": [[5, "omit-nucleotide-at-cross-linking-site"]], "Maximum number of motifs": [[5, "maximum-number-of-motifs"]], "Minumum width of motifs": [[5, "minumum-width-of-motifs"]], "Maximum width of motifs": [[5, "maximum-width-of-motifs"]]}, "indexentries": {}})